modules/serverless-services-network-policies.adoc

```diff
@@ -0,0 +1,82 @@
+[id="serverless-services-network-policies_{context}"]
+= Enabling communication with Knative applications on a cluster with restrictive network policies
+
+If you are using a cluster which multiple users have access to, your cluster might use network policies to control which pods, services, and namespaces can communicate with each other over the network
+
+If your cluster uses restrictive network policies, it is possible that Knative system pods are not able to access your Knative application. For example, if your namespace has the following network policy, which denies all requests, Knative system pods cannot access your Knative application:
+
+.Example NetworkPolicy object that denies all requests to the namespace
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: deny-by-default
+  namespace: example-namespace
+spec:
+  podSelector:
+  ingress: []
+...
+----
+
+To allow access to your applications from Knative system pods, you must add a label to each of the Knative system namespaces, and then create a `NetworkPolicy` object in your application namespace that allows access to the namespace for other namespaces that have this label.
+
+[IMPORTANT]
+====
+A network policy that denies requests to non-Knative services on your cluster still prevents access to these services. However, by allowing access from Knative system namespaces to your Knative application, you are allowing access to your Knative application from all namespaces in the cluster.
+
+If you do not want to allow access to your Knative application from all namespaces on the cluster, you might want to use _JSON Web Token authentication for Knative services_ instead (see the _Knative Serving_ documentation). JSON Web Token authentication for Knative services requires Service Mesh.
+====
+// xrefs for modules would be nice here to link to the JWT docs
+
+.Procedure
+
+. Add the `knative.openshift.io/system-namespace=true` label to each Knative system namespace that requires access to your application:
++
+.Label the `knative-serving` namespace
```

Comment:
Just wondering if each of these would be better as substeps (1a, 1b, etc.) instead of stacked code blocks?

Reply:
Really good suggestion, not sure why I didn't do this in the first place 😄 thank you!